### ISSUE TITLE
Read Hardware Config from File

### ISSUE THREAD

**Issue by brandonrobertz** [FourThievesVinegar/microlab#3]
<a href="https://github.com/TheDukeZip"><img src="https://avatars.githubusercontent.com/u/3036495?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [TheDukeZip](https://github.com/TheDukeZip)**
_Sunday Aug 21, 2016 at 02:15 GMT_
_Originally opened as https://github.com/TheDukeZip/4tvc/issues/3_

----

The services should pick up the hardware config (such as the number of syringes, and which GPIO pins are associated with the motors and the switches) from one config file.

This allows us to have one place to configure the hardware GPIO pins (later on, we could have a UI that allows the user to configure them.) 

It also allows us to have a variable number of each hardware device. You could add another syringe to the config and the service would automatically show it as available, and it could be controlled through the interface.

We should probably do this after #2 so that we know exactly what properties each device contains.

We also may want to speak to #Chemistry about whether we would ever need to support two heaters running off the readings from one temp sensor... or using two temp sensors with one heating element.

**Comment by brandonrobertz** [comment]
<a href="https://github.com/TheDukeZip"><img src="https://avatars.githubusercontent.com/u/3036495?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [TheDukeZip](https://github.com/TheDukeZip)**
_Sunday Aug 21, 2016 at 02:56 GMT_

----

@MichaelSLaufer mentioned "Looking at the concerns on Github, I just want to add that the only time we would need to support multiple heaters or temp sensors is if we were running a volume of liquid so large that we were concerned that the stirring wasn't sufficient to eliminate temperature gradients in the vessel. It seems unlikely. " "Of course aesthetically, it's always nice to have things be expandable, but I don't think it's necessary. And less so at this stage."
